datasette/database.py

```diff
@@ -152,9 +152,17 @@ def in_thread():
                 setattr(connections, self.name, conn)
             return fn(conn)
 
-        return await asyncio.get_event_loop().run_in_executor(
-            self.ds.executor, in_thread
-        )
+        executor = asyncio.get_event_loop().run_in_executor(self.ds.executor, in_thread)
+        try:
+            return await asyncio.wait_for(
+                executor,
+                timeout=(time_limit / 1000.0) if time_limit is not None else None,
+            )
+        except asyncio.TimeoutError:
+            conn = getattr(connections, self.name, None)
+            if conn:
+                conn.interrupt()
```

Comment:
This will never work, because `connections` is a thread local: https://github.com/simonw/datasette/blob/fb2ad7ada0b86a7fe4a576fe23236757c41eb05e/datasette/database.py#L24